src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/ChooseShippingMethodHandler.php

```diff
@@ -66,6 +67,10 @@ public function __invoke(ChooseShippingMethod $chooseShippingMethod): OrderInter
 
         $stateMachine = $this->stateMachineFactory->get($cart, OrderCheckoutTransitions::GRAPH);
 
+        if ($cart->getShippingAddress() === null) {
+            throw new OrderCannotBeShippedWithoutAddressing();
```

Comment:
I wonder, is it something we would like to do with a custom exception? Or maybe it should be a guard on state machine? 🤔 We're checking the possibility of transitioning order below nonetheless. cc @GSadee @lchrusciel

Reply:
state machine is check in the next lines, it throws 500 error if state cant be applied, this way we capture error we expect before and throw custom error to tell what caused this error.

Reply:
I totally understand that 😄 I just wonder shouldn't we keep the order validation in one place rather than have it done with two fashions (custom exception + state machine checking). Still, it's just an improvement 🖖

Reply:
Validation should be handled mostly with the validation component. That should be possible with the command, that we are having here. I'm wondering if we should start moving some protecting rules into the state machine, as we don't have any yet. Exception itself is not a big deal in the current implementation tho

Reply:
https://github.com/Sylius/Sylius/pull/12746 okay, added another approach (this time with validations)